src/vmod_xkey.c

```diff
@@ -628,6 +641,7 @@ vmod_event(VRT_CTX, struct vmod_priv *priv, enum vcl_event_e e)
 #endif
 			AZ(xkey_cb_handle);
 			xkey_cleanup();
+			VSC_xkey_Destroy(&vsc_seg);
```

Comment:
... and here vsc should be set to NULL

Reply:
It's null'd by the destroy function.

Reply:
I don't think the destroy function has a reference to `vsc`.

Reply:
My bad, mixed up with vsc_seg.

Reply:
If the destroy function were to return NULL...

Reply:
> If the destroy function were to return NULL...

That could be nice. Then the API would be the same for New and Destroy.

```c
vsc = VSC_xkey_New(...);
vsc = VSC_xkey_Destroy(...);
```

I added:

```c
vsc = NULL;
```